support-frontend/assets/pages/[countryGroupId]/thank-you.tsx

```diff
@@ -82,13 +106,52 @@ export function ThankYou({ geoId }: Props) {
 		return <div>Unable to find contribution type {contributionType}</div>;
 	}
 
-	const isOneOffPayPal =
-		order.paymentMethod === 'PayPal' && order.product === 'Contribution';
-
 	const abParticipations = abTestInit({ countryId, countryGroupId });
 	const showOffer =
 		!!abParticipations.usFreeBookOffer && order.product === 'SupporterPlus';
 
+	const isOneOff = order.product === 'Contribution';
+	const isOneOffPayPal = order.paymentMethod === 'PayPal' && isOneOff;
+	const isSupporterPlus = order.product === 'SupporterPlus';
+	const isNewAccount = order.userTypeFromIdentityResponse === 'new';
+	const csrf = { token: order.csrf };
+
+	const thankYouModuleData = getThankYouModuleData(
+		countryId,
+		countryGroupId,
+		csrf,
+		order.email,
+		isOneOff,
+		isSupporterPlus,
+	);
+	const maybeThankYouModule = (
+		condtion: boolean,
+		moduleType: ThankYouModuleType,
+	): ThankYouModuleType[] => (condtion ? [moduleType] : []);
+
+	const thankYouModules: ThankYouModuleType[] = [
+		...maybeThankYouModule(isNewAccount && !isSignedIn, 'signUp'),
+		...maybeThankYouModule(
+			!isNewAccount && !isSignedIn && order.email.length > 0,
+			'signIn',
+		),
+		...maybeThankYouModule(
+			contributionType !== 'ONE_OFF' && isSupporterPlus,
+			'appDownload',
+		),
+		...maybeThankYouModule(
+			contributionType === 'ONE_OFF' && order.email.length > 0,
+			'supportReminder',
+		),
+		...maybeThankYouModule(order.email.length > 0, 'feedback'),
+		...maybeThankYouModule(countryId === 'AU', 'ausMap'),
+		'socialShare',
+	];
+
+	const numberOfModulesInFirstColumn = thankYouModules.length >= 6 ? 3 : 2;
+	const firstColumn = thankYouModules.slice(0, numberOfModulesInFirstColumn);
+	const secondColumn = thankYouModules.slice(numberOfModulesInFirstColumn);
```

Comment:
Do you think this API into the Thank you modules is clear? Just wondering if there's an opportunity here for a refactor that might make this clearer to folks in the future.

Reply:
Personally, here, I think its pretty clear how the Thankyou components are split into columns depending on the total number of thankyouModules shown.